src/backend/distributed/metadata/node_metadata.c

```diff
@@ -1343,7 +1344,7 @@ citus_update_node(PG_FUNCTION_ARGS)
 		}
 	}
 
-	WorkerNode *workerNode = LookupNodeByNodeId(nodeId);
+	WorkerNode *workerNode = FindNodeAnyClusterByNodeId(nodeId);
```

Comment:
I hit an assertion below:
```C
  1422		/* we should be able to find the new node from the metadata */
   1423		workerNode = FindWorkerNode(newNodeNameString, newNodePort);
-> 1424		Assert(workerNode->nodeId == nodeId);
   1425	
```

I guess we should use `FindWorkerNodeAnyCluster` instead of `FindWorkerNode`

Reply:
Good catch!